src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+why.
+
+Also, there are times when programmers need to isolate or ignore certain
+portions of their code as they are testing it. In the "Try It" box above, you
+were instructed to *remove* a line of code in order to create compiler errors.
+However, programmers are usually reluctant to delete lines that they might need
+to bring back.
+
+.. index:: ! comments
+
+Best practice encourages us to add **comments** to our programs. These are
+notes that clearly explain what the code is doing.
+
+A comment is text within a program intended only for a human reader—--it is
+completely ignored by the compiler. In Java, the ``//`` token indicates the
+start of a comment, and the rest of the line gets ignored. For comments that
+stretch over multiple lines, the text falls between the symbols ``/*   */``.
+
+Comments can also be used to temporarily skip a portion of the code when a
+program runs. Instead of removing line 6 in ``TempConverter``, we could
+*comment out* the line. This would create the same compiler errors we wanted to
+witness, but it would preserve the original code and allow us to easily
+reactivate it by removing the ``//`` token from the line.
+
+.. admonition:: Example
+
+   .. sourcecode:: Java
+      :linenos:
+
+      import java.util.Scanner;
+
+      // Here is an example of a comment.
+
+      /* Here is how
+      to have
+      multi-line
+      comments. */
+
+      /*
+      Or
+      like
+      this.
+      */
+
+      public class HelloWorld {
+         public static void main(String[] args) {
+            Scanner input; // Comments do not have to start at the beginning of a line.
+
+            input = new Scanner(System.in);
+            System.out.println("Please enter your first name: ");
+            String name = input.next(); //Declares the 'name' variable and initializes it with text from the command line.
+            input.close();
+
+            System.out.println("Hello, " + name + "!");
+
+            // System.out.println("This line will NOT print!");
+         }
+      }
+
+Collect Input with the ``Scanner`` Class
+-----------------------------------------
+
+In Java, ``Scanner`` objects make getting input from the user, a file, or even
+over the network relatively easy. For our temperature conversion program, we
+declared the variable ``input`` to be of type ``Scanner``.
+
+.. sourcecode:: java
+   :lineno-start: 7
+
+   Scanner input;
+
+We want our program to prompt the user to enter in a number in the command
+line. We accomplish this by creating a ``Scanner`` instance using the word
+``new`` and then calling the constructor and passing it the ``System.in``
+object:
+
+.. sourcecode:: java
+   :lineno-start: 9
+
+   input = new Scanner(System.in);
+
+Notice that this ``Scanner`` object is assigned to the name ``input``, which we
+declared to be a ``Scanner`` object earlier. ``System.in`` is similar to
+``System.out`` except, as the name implies, it is used for input.
```

Comment:
well said!

Reply:
Thanks.  Again, compliments to the original author.